packages/hooks/src/useCollectionQuery/useCollectionQuery.mdx

```diff
@@ -0,0 +1,143 @@
+---
+name: useCollectionQuery
+menu: Hooks
+route: /hooks/use-collection-query
+---
+
+import { Playground } from "docz";
+import { List } from "@jobber/components/List";
+import { Button } from "@jobber/components/Button";
+import { Spinner } from "@jobber/components/Spinner";
+import { Divider } from "@jobber/components/Divider";
+import { useCollectionQuery } from "@jobber/hooks";
+import {
+  apolloClient,
+  LIST_QUERY,
+  mapDataToListItem,
+  styles,
+  propsList,
+  subscriptionPropsList,
+} from "./mdxUtils";
+
+# UseCollectionQuery
+
+`useCollectionQuery` is a generalized hook that that fetches a collection of
+data. For use with any lists populated by graphQL collections.
+
+```tsx
+import { useCollectionQuery } from "@jobber/hooks";
+
+const {
+  data,
+  error,
+  refresh,
+  nextPage,
+  loadingRefresh,
+  loadingNextPage,
+  loadingInitialContent,
+} = useCollectionQuery<ListQueryType, SubscriptionType>({
+  query: LIST_QUERY,
+  queryOptions: {
+    fetchPolicy: "network-only",
+    nextFetchPolicy: "cache-first",
+  },
+  getCollectionByPath(items) {
+    return items.path.to.collection;
+  },
+  subscription: {
+    // subscriptions are optional
+    document: SUBSCRIPTION_QUERY,
+    getNodeByPath(items) {
+      return items.path.to.node;
+    },
+});
+```
+
+<Playground>
+  {() => {
+    const {
+      data,
+      error,
+      refresh,
+      nextPage,
+      loadingRefresh,
+      loadingNextPage,
+      loadingInitialContent,
+    } = useCollectionQuery({
+      // useCollectionQuery should be called with the query type, and
+      // optionally, the subscription type. The playground errors with
+      // typing included, so typing has been removed in this example.
+      // Please see the first example for appropriate typing.
+      query: LIST_QUERY,
+      queryOptions: {
+        fetchPolicy: "network-only",
+        nextFetchPolicy: "cache-first",
+        client: apolloClient,
+      },
+      getCollectionByPath(items) {
+        return items.allPlanets;
+      },
+    });
+    const items = mapDataToListItem(data) || [];
+    return (
+      <>
+        <div style={styles.spinners}>
+          <div style={styles.loadingIndicator}>
+            <h4>Initial Loading Indicator</h4>
+            {loadingInitialContent && <Spinner />}
+          </div>
+          <div style={styles.loadingIndicator}>
+            <h4>Refreshing Indicator</h4>
+            {loadingRefresh && <Spinner />}
+          </div>
+          <div style={styles.loadingIndicator}>
+            <h4>Fetching More Indicator</h4>
+            {loadingNextPage && <Spinner />}
+          </div>
+        </div>
```

Comment:
It's a bit of an abuse of intention but, it might be easier to just use a `Button` component with the `disabled` prop here rather than a bunch of custom styling. Just thinking if we can strip out some of the boiler plate it can better keep the example focused on the workings of the hook itself.

@chris-at-jobber thoughts?

Reply:
@seanhealy This is an interesting idea. I am curious what @chris-at-jobber thinks. Would these buttons not be functional other then being disabled / enabled?

Reply:
I am not sure I understand what is being suggested, or why 😅  What would the buttons replace? Why would they be disabled?

Reply:
@chris-at-jobber My interpretation is that there would be three buttons that did nothing when clicked. These buttons would be disabled when not in a loading state, and enabled if they were in a loading state. The code becomes much simpler, and does not need styling. I actually really like this change! I have recorded a demo for you :) 


https://user-images.githubusercontent.com/6353164/123437099-52c29e00-d59d-11eb-8773-d040ffa8aa40.mp4

Reply:
Hmmm

I don't think I understand what those buttons are actually communicating, and they flicker in and out of "active" to quickly to be of any real help. I thought the previous approach of "header + spinner" was being used to visualize that different loading paths are hit in the code for different types of loading, but if that's not what we're trying to do, maybe just remove the whole section? It seems like the list content updating when you trigger the actions is good enough.